[RUM][Mobile] RUMM-2892 iOS SDK `2.0` Release (#19061)
DataDog/documentation: content/en/real_user_monitoring/ios/advanced_configuration.md

> In addition to [tracking actions automatically](#automatically-track-user-actions), you can track specific custom user actions (taps, clicks, and scrolls) with the `addUserAction(type:name:)` API. 

Reviewer: ```suggestion
In addition to [tracking actions automatically](#automatically-track-user-actions), you can track specific custom user actions (taps, clicks, and scrolls) with the `addAction(type:name:)` API. 
```

> To manually register instantaneous RUM actions such as `.tap` on `RUMMonitor.shared()`, use `.addUserAction(type:name:)`. For continuous RUM actions such as `.scroll`, use `.startUserAction(type:name:)` or `.stopUserAction(type:)`.

Reviewer: Ref.: _breaking changes_ in https://github.com/DataDog/dd-sdk-ios/pull/1310
```suggestion
To manually register instantaneous RUM actions such as `.tap` on `RUMMonitor.shared()`, use `.addAction(type:name:)`. For continuous RUM actions such as `.scroll`, use `.startAction(type:name:)` or `.stopAction(type:)`.
```

For example:

{{< tabs >}}
{{% tab "Swift" %}}
```swift
import DatadogRUM

// in your `UIViewController`:

let rum = RUMMonitor.shared()

@IBAction func didTapDownloadResourceButton(_ sender: UIButton) {

>     rum.addUserAction(

Reviewer: ```suggestion
    rum.addAction(
```

### Custom Resources

In addition to [tracking resources automatically](#automatically-track-network-requests), you can also track specific custom resources such as network requests or third-party provider APIs. Use the following methods on `RUMMonitor.shared()` to manually collect RUM resources:

- `.startResource(resourceKey:request:)`
- `.stopResource(resourceKey:response:)`
- `.stopResourceWithError(resourceKey:error:)`

> - `.stopResourceWithError(resourceKey:errorMessage:)`

Reviewer: ```suggestion
- `.stopResourceWithError(resourceKey:message:)`
```
Ref.: https://github.com/DataDog/dd-sdk-ios/pull/1310
> `errorMessage` parameter in `stopResourceWithError()` was renamed to `message` - for simplicity

For example:

{{< tabs >}}
{{% tab "Swift" %}}
```swift
import DatadogRUM

// in your network client:

let rum = RUMMonitor.shared()

> rum.startResourceLoading(

Reviewer: ```suggestion
rum.startResource(
```
Ref.: https://github.com/DataDog/dd-sdk-ios/pull/1310
> resource tracking APIs are renamed - we no longer use word "Loading":
> - `startResourceLoading()` → `startResource()`
> - `stopResourceLoading()` → `stopResource()`
> - `stopResourceLoadingWithError()` → `stopResourceWithError()`
> this is to align with `dd-sdk-android` where we never used "Loading" phrase

    resourceKey: "resource-key", 
    request: request
)

> rum.stopResourceLoading(

Reviewer: ```suggestion
rum.stopResource(
```

### Custom Resources

In addition to [tracking resources automatically](#automatically-track-network-requests), you can also track specific custom resources such as network requests or third-party provider APIs. Use the following methods on `RUMMonitor.shared()` to manually collect RUM resources:

- `.startResourceLoading(resourceKey:request:)`

> - `.stopResourceLoading(resourceKey:response:)`

Reviewer: Here and more in other snippets:
> resource tracking APIs are renamed - we no longer use word "Loading":
> - startResourceLoading() → startResource()
> - stopResourceLoading() → stopResource()
> - stopResourceLoadingWithError() → stopResourceWithError()

Ref. _breaking changes_ from https://github.com/DataDog/dd-sdk-ios/pull/1310

```objective-c
// in your network client:

> [[DDRUMMonitor shared] startResourceLoadingWithResourceKey:@"resource-key"
>                                                    request:request
>                                                 attributes:@{}];

Reviewer: ```suggestion
[[DDRUMMonitor shared] startResourceWithResourceKey:@"resource-key"
                                            request:request
                                         attributes:@{}];
```

> [[DDRUMMonitor shared] stopResourceLoadingWithResourceKey:@"resource-key"
>                                                  response:response
>                                                attributes:@{}];

Reviewer: ```suggestion
[[DDRUMMonitor shared] stopResourceWithResourceKey:@"resource-key"
                                          response:response
                                        attributes:@{}];
```

## Initialization Parameters
 

> You can use the following properties in `Datadog.Configuration` when creating the Datadog configuration to initialize the library:

Reviewer: ```suggestion
You can use the following properties in `RUM.Configuration` when creating the Datadog configuration to initialize the library:
```

`site`
: Sets the Datadog server endpoint that data is sent to.

`batchSize`
: Sets the preferred size of batched data uploaded to Datadog. This value impacts the size and number of requests performed by the RUM iOS SDK (small batches mean more requests, but each request becomes smaller in size). Available values include: `.small`, `.medium`, and `.large`.

`uploadFrequency`
: Sets the preferred frequency of uploading data to Datadog. Available values include: `.frequent`, `.average`, and `.rare`.

### RUM configuration

You can use the following properties in `RUM.Configuration` when enabling RUM:

`sessionSampleRate`
: Sets the sampling rate for RUM sessions. The `sessionSampleRate` value must be between `0.0` and `100.0`. A value of `0.0` means no sessions are sent, `100.0` means all sessions are sent to Datadog. If not configured, the default value of `100.0` is used.

`uiKitViewsPredicate`
: Enables tracking `UIViewControllers` as RUM views. You can use default implementation of `predicate` by setting the `DefaultUIKitRUMViewsPredicate` or implement [your own `UIKitRUMViewsPredicate`](#automatically-track-views) customized for your app.

`uiKitActionsPredicate`

> : Enables tracking user interactions (taps) as RUM actions. You can use the default implementation of `predicate` by setting the `DefaultUIKitRUMActionsPredicate` or implement [your own `UIKitRUMUserActionsPredicate`](#automatically-track-user-actions) customized for your app.

Reviewer: ```suggestion
: Enables tracking user interactions (taps) as RUM actions. You can use the default implementation of `predicate` by setting the `DefaultUIKitRUMActionsPredicate` or implement [your own `UIKitRUMActionsPredicate`](#automatically-track-user-actions) customized for your app.
```

If your app is running on devices behind a custom proxy, you can inform the RUM iOS SDK's data uploader to ensure that all tracked data is uploaded with the relevant configuration. 

When initializing the iOS SDK, specify this in your proxy configuration.

{{< tabs >}}
{{% tab "Swift" %}}
```swift
import DatadogCore

Datadog.initialize(
  with: Datadog.Configuration(
    clientToken: "<client token>",
    env: "<environment>",
    proxyConfiguration: [
        kCFNetworkProxiesHTTPEnable: true, 
        kCFNetworkProxiesHTTPPort: 123, 
        kCFNetworkProxiesHTTPProxy: "www.example.com", 
        kCFProxyUsernameKey: "proxyuser", 
        kCFProxyPasswordKey: "proxypass" 
    ]
  ), 

>   trackingConsent: .trackingConsent

Reviewer: ```suggestion
  trackingConsent: trackingConsent
```
